### DISCUSSION TITLE
Select all columns

### DISCUSSION THREAD

**Discussion by Sytten** [SeaQL/sea-query#101] [Ideas]
Hi!
I could not find an easy way to specify all columns (but not star).
Maybe Iden could implement an `all()` function that returns a vec?
It would look like:
```rust
Query::select().columns(MyModel::all())
```
Thanks!

**Comment by Progdrasil** [comment]
Since `columns` is generic on `IntoIterator` which all iterators implement, you could simply use the [`EnumIter`](https://docs.rs/strum_macros/0.21.1/strum_macros/derive.EnumIter.html) derive from [strum](https://crates.io/crates/strum) crate which will give you a `iter()` method on your enum which is the equivalent to your `all`

```rust
use strum::EnumIter;
use sea_query::{Iden, Query};

#[derive(Iden, EnumIter)]
enum MyModel {
    Id,
    MyColumn,
    /* etc */
}

Query::select().columns(MyModel::iter())
```

**Comment by Sytten** [comment]
Will use it, still think it would be a good idea to have the `all()` since `Iden` can be applied to structs too.

**Comment by Progdrasil** [comment]
An Iden implementation on a struct will only give a single column reference. If you wish to associate multiple iden's to a single struct you're still going to need an enum and associate it to the struct. If you want this to all be automatically derived, you're looking for an ORM.

**Comment by Sytten** [comment]
Right that make sense. I usually prefer query builders. The Iden on struct could generate Iden for all fields too or generate a DSL but I guess that would be more for an ORM.

**Comment by Progdrasil** [comment]
I think thats also in the ORM territory, since you still need to generate a list of something that implements iden. Since iden cannot give a stringified list  because the query builder quotes the output of that string. You could make that it returns a list of  `Alias` struct.

**Comment by tyt2y3** [comment]
@Progdrasil thank you for helping out
@Sytten we are releasing our ORM soon (in this month), so if you are interested we'll let you know

**Comment by Sytten** [comment]
@tyt2y3 I am most likely going to continue using the query builder because I have complex queries with a lot of conditionals and joins that are usually very hard to express in an ORM (thus our switch from diesel to sea-query).

**Comment by tyt2y3** [comment]
Now that [SeaORM](https://github.com/SeaQL/sea-orm) is 'public', please take a look and see if it helps!
You still have full access to the SeaQuery query builder when you need it.
Please close this issue if that works for you.
